### PR TITLE
[#1802, #1803, #1804, #1876] Added HTML App Menu for Windows

### PR DESCRIPTION
**Pull request by tonyanziano**
#1876, #1802, #1803, #1804

===

image

image

image

### CODE REVIEW

File: packages/sdk/ui-react/src/widget/menu/menuButton.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  menuShowing: boolean;
+}
+
+/* Button that will spawn a menu when clicked */
+export class MenuButton extends React.Component<MenuButtonProps, MenuButtonState> {
+  private static _id: number;
+  private buttonRef: HTMLButtonElement;
+  private menuButtonId: string;
+  private onBodyClickListener: () => void;
+  private onMenuItemSelectedListener: () => void;
```
Comment (corinagum) [inline packages/sdk/ui-react/src/widget/menu/menuButton.tsx]: lines 57 and 58 alphabetize?

File: packages/sdk/ui-react/src/widget/menu/menu.scss [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      color: var(--menu-item-disabled-color);
+      outline: none;
+    }
+
+    > .submenu-caret {
+      background-color: var(--menu-item-hover-color);
+    }
+
+    > .menu-item-check {
+      background: var(--menu-item-hover-color);
```
[the remark this answers is not in the document]
Comment (tonyanziano) [reply packages/sdk/ui-react/src/widget/menu/menu.scss]: good catch!

File: packages/app/client/src/ui/shell/appMenu/appMenuTemplate.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+          webContents.getZoomFactor(zoomFactor => {
+            const newZoomFactor = zoomFactor - 0.1;
+            if (newZoomFactor <= minZoomFactor) {
+              webContents.setZoomFactor(minZoomFactor);
+            } else {
+              webContents.setZoomFactor(newZoomFactor);
+            }
+          });
+        },
+        subtext: `${CtrlOrCmd}+-`,
```
[the remark this answers is not in the document]
Comment (tonyanziano) [reply packages/app/client/src/ui/shell/appMenu/appMenuTemplate.ts]: Honestly, I'm not entirely sure. Like you pointed out, the keybind already existed like that so I just carried it over.

[the remark this answers is not in the document]
Comment (corinagum) [reply packages/app/client/src/ui/shell/appMenu/appMenuTemplate.ts]: Looking at VS Code, both `Ctrl/Cmd =` and `Ctrl/Cmd Shift =` are zoom in, and likewise Ctrl/Ctrl Shift for - are zoom out. Should Emulator be updated to the same?

[the remark this answers is not in the document]
Comment (tonyanziano) [reply packages/app/client/src/ui/shell/appMenu/appMenuTemplate.ts]: Added shortcuts for both. 👍

**Comment by tonyanziano** [comment]
@corinagum Addressed your comments.

Thanks for the note on the menu accessibility, I didn't even know you could use `Ctrl+<arrow>` to navigate lists, so I didn't test any of that.

My thought is that we submit it as is, and then if the a11y team ends up dinging us for that, we can fix it. However, to my memory, I don't think they reported any issues involving `Ctrl+<arrow>` navigation, but maybe we just didn't have any instances like the app menu.

**Pull request merged**